Snabble/Core/Payment/PaymentMethodDetails.swift

```diff
@@ -144,6 +149,43 @@ public enum CreditCardBrand: String, Codable {
     }
 }
 
+// response data from the telecash IPG Connect API
+struct ConnectGatewayResponse {
+    var hostedDataId: String { data["hosteddataid"] ?? "" }
+    var schemeTransactionId: String { data["schemeTransactionId"] ?? "" }
+    var cardNumber: String { data["cardnumber"] ?? "" }
+    var cardHolder: String { data["bname"] ?? "" }
+    var brand: String { data["ccbrand"] ?? "" }
+    var expMonth: String { data["expmonth"] ?? "" }
+    var expYear: String { data["expyear"] ?? "" }
+    var responseCode: String { data["processor_response_code"] ?? "" }
+    var failReason: String { data["fail_reason"] ?? "" }
+    var failCode: String { data["fail_rc"] ?? "" }
+    var orderId: String { data["oid"] ?? "" }
+
+    private var data = [String: String]()
+
+    init(response: [[String: String]]) {
```

Comment:
Was hältst du davon das Objekt in der Init zu prüfen und somit das `isValid` zu vermeiden? Vereinfacht das Arbeiten mit der Instanz, da klar ist wenn es die Instanz gibt ist es gültig.

Ansonsten würde ich gerne auch `empty` und `nil` nicht vermischen. Siehst du dazu einen Mehrwert? Ich empfinde es sehr schwierig, da jeder Entwickler das Kontext wissen haben muss bzgl. `empty` und `nil`. Wir haben eine Sprache die Optionals ermöglicht, daher würde ich vermuten das die Mehrheit der Entwickler auch davon ausgesehen das es verwendet wird.

Reply:
Aus dem Javascript-Event können im Fehlerfall sowohl "" als auch nil ankommen, das hier ist dazu da dass wir das nur einmal einheitlich behandeln müssen. Wir können das aber gerne in einen failable init packen, damit das von aussen niemand mehr sieht, klar.

Reply:
Wobei das auch blöd ist. Wir müssen nämlich auch noch den Fall des Abbrechens durch den User abhandeln, und das geht nur wenn man einen Property der Response mit "5993" vergleicht...

Reply:
Wieviele Varianten gibt es? Macht es eventuell Sinn mit einem Protocol und konkreten Implementierungen zu arbeiten? Ich meine damit für jeden behandelten Response-Code eine konkrete Implementierung. Für den Fall das ein Response-Code kommt den wir _noch_ nicht verstehen wird ein Error in der Throwable init geworfen.  Würde vor allem unseren zukünftigen Ichs das Verständnis vereinfachen, da im Code die Response-Codes bereits klar aufgeteilt sind und klar ist welche Werte bei welchen Codes notwendig sind. 🤔 Was meinst du?

Reply:
Im Moment gibt's nur "00" und "5993" die eine spezielle Bedeutung haben.